Source/DafnyLanguageServer/Language/Symbols/FunctionSymbol.cs

```diff
@@ -8,7 +9,23 @@ public class FunctionSymbol : MemberSymbol, ILocalizableSymbol {
 
     public IList<VariableSymbol> Parameters { get; } = new List<VariableSymbol>();
 
-    public override IEnumerable<ISymbol> Children => Parameters;
+    /// <summary>
+    /// Gets the body of the function
+    /// </summary>
+    public ScopeSymbol? Body { get; set; }
+    public ScopeSymbol? ByMethodBody { get; set; }
```

Comment:
These are many new properties but I've only seen their declaration.
Did you introduce these to extend the document outline view in visual studio code? Otherwise, it's unnecessary since only the lookup tree is relevant for hover information. If you did this for the document outline, I'd prefer to have this in a separate PR to better judge their use.

Reply:
I don't know what you mean by "document outline" but I did add these properties to be able to account for the new expression scopes that can introduce variables in every of these expressions.

Reply:
ByMethodBody is a recent language feature, which enables to provide and prove an compile-time implementation for a given function.
For example:
```
predicate even(n: nat)
  ensures even(n) <==> n % 2 == 0 
{
  if n < 2 then n == 0 else even(n - 2)
} by method {
  var x := n % 2 == 0;
  return x;
}
```

I really prefer to keep the same name as the AST defines it. The language may evolve, and maybe functions will start to support statements instead of expressions, so we want to be as conservative as possible. I'm ok expanding names from the original AST but it eases the learning curve and the debugging curve if the names match.